AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -7,25 +7,69 @@
 //
 
 import UIKit
+import MobileCoreServices
+import AVKit
+import AVFoundation
 
-class DoubleVideoViewController: UIViewController, CellTitled {
+class DoubleVideoViewController: UIViewController, CellTitled, UIImagePickerControllerDelegate,UINavigationControllerDelegate {
+    
+    var movieURL: URL?
+    
     var titleForCell: String = "Double Video"
-
+    
+    let picker = UIImagePickerController()
+    
     @IBOutlet weak var videoContainerTop: UIView!
     @IBOutlet weak var videoContainerBottom: UIView!
     
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        picker.delegate = self
+        
         // don't f with this line
         self.edgesForExtendedLayout = UIRectEdge(rawValue: 0)
     }
-
+    
     override func viewDidLayoutSubviews() {
         // update your layers' frames here
     }
     
     @IBAction func loadVideo(_ sender: UIButton) {
+        picker.sourceType = .photoLibrary
+        picker.mediaTypes = UIImagePickerController.availableMediaTypes(for: .photoLibrary)!
+        present(picker, animated: true, completion: nil)
+    }
+
     
+    
+    //MARK: - Delegates
+    private func imagePickerController(_ picker: UIImagePickerController,didFinishPickingMediaWithInfo info: [String : AnyObject]) {
```

Comment:
The main issue here, and why you couldn't show video was that this delegate method was defined incorrectly. It had the wrong type for info; should be [String: Any] and it should not be private. This is the correct signature for the function:

```swift
 func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any])
```
Since delegate methods are called by something else it's easy for the compiler to miss this error. You made a valid function but it was not reachable by the calling object because the types were different *and* because, being private no class other than this one can call the method.